src/app/Http/Controllers/PostController.php

```diff
@@ -95,10 +101,15 @@ public function destroy(Post $post): RedirectResponse
     /**
      * Rate a post.
      */
-    public function rate(Post $post, int $rating)
+    public function rate(Post $post, int $rating): RedirectResponse
     {
         $post->rateOnce($rating, null, Auth::id());
 
+        // Send warning notification to the admin if this post has received a lot of low ratings.
+        if ($post->averageRating() < $this::WARNING_THRESHOLD_RATING && $post->timesRated() >= $this::WARNING_THRESHOLD_NUMBER) {
+            sendWarningNotification($post->title);
+        }
+
```

Comment:
Will the admin continuously receive emails about the same post?

Reply:
Oh, that's a very good point. Yes, he will. Ideally, we would store that a warning has already been sent and send reminders after some time. But for now, I think we can keep it like this, since it's just a prototype and we have other priorities.